src/BlockchainTestsFiller/ValidBlocks/bcEIP3675/PreviousMinerBalanceFiller.yml

```diff
@@ -30,26 +30,52 @@ PreviousMinerBalance:
       code: :raw 0x60FF00
       storage: {}
 
+  # Contract with 0 gas code execution.
+    0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb:
+      balance: 0x10000000000
+      nonce: 1
+      code: ''
+      storage: {}
+
     0xd02d72E067e77158444ef2020Ff2d325f929B363:
       balance: 0x10000000000000000
       nonce: 1
       code: ''
       storage: {}
 
-    #Store the beneficiary miner balance (coinbase) after blocks mined.
+    #Store miner tip at block finalization.
     0xcccccccccccccccccccccccccccccccccccccccc:
       balance: 0x10000000000
       nonce: 1
       code: |
-         :yul { sstore(sub(number(),1), sload(0xFFFF) )
-                if eq(calldatasize(), 0) { sstore(0xFFFF, balance(coinbase())) }
-                if gt(calldatasize(), 0) { sstore(0xFFFF, 0) }
+         :yul {
+               let prevoiusBlockNumber := sub(number(),1)                                        //Miner tips calculated the block after the tip, so go back one block.
+               let fixedStorageSlotMinerBalance := 0xFFFF                                        //Storage slot for current block to be used in the next block as previous balance.
+               let storageSlotMinerValue :=  add(0x10000,prevoiusBlockNumber)                    //Storage slot for miner balance every block.
+               let storageSlotMinerCheck :=  add(0x20000,prevoiusBlockNumber)                    //Storage slot to calculate if current balance is valid based on current tip and previous balance.
+               let currentMinerBalance := balance(coinbase())                                    //Get the current miner balance state.
+               let previousMinerBalance := sload(fixedStorageSlotMinerBalance)                   //Get the previous miner balance state.
+               let minerTipCurrentBlock := sub(currentMinerBalance, previousMinerBalance)        //Calculate current miner tip.
+               let previousMinerBalancePlusTip := add(previousMinerBalance,minerTipCurrentBlock) //Add current tip and previous miner balance.
+               let isCurrentBalanceValid := eq(currentMinerBalance,previousMinerBalancePlusTip)  //Return 1 if current balance equals current tip + previous balance.
+               sstore(prevoiusBlockNumber, minerTipCurrentBlock)                                 //Store the current miner tip.
```

Comment:
it is not minerTip of the current block, the variable name misleading. 
sub(currentMinerBalance, previousMinerBalance)   would have then diff that miner received with finalization of previous block. that include tips and reward.